python_modules/libraries/dagster-aws/dagster_aws/s3/resources.py

```diff
@@ -1,77 +1,107 @@
-from dagster import Field, Noneable, StringSource, resource
-from dagster._utils.merger import merge_dicts
+from typing import Any, Optional, TypeVar
+
+from dagster import (
+    ConfigurableResourceFactory,
+    InitResourceContext,
+    resource,
+)
+from pydantic import Field
 
 from .file_manager import S3FileManager
 from .utils import construct_s3_client
 
-S3_SESSION_CONFIG = {
-    "use_unsigned_session": Field(
-        bool,
-        description="Specifies whether to use an unsigned S3 session",
-        is_required=False,
-        default_value=False,
-    ),
-    "region_name": Field(
-        str, description="Specifies a custom region for the S3 session", is_required=False
-    ),
-    "endpoint_url": Field(
-        StringSource,
-        description="Specifies a custom endpoint for the S3 session",
-        is_required=False,
-    ),
-    "max_attempts": Field(
-        int,
+T = TypeVar("T")
+
+
+class S3ResourceBase(ConfigurableResourceFactory[T]):
```

Comment:
same discussion to be had as the upstream diff about these "resource base" classes

Reply:
Updated to `ResourceWithS3Configuration`, for now